Automattic/wp-calypso
# Refactor G Suite Purchase CTA

#### Changes proposed in this Pull Request

* New Sub Component For Rendering G Suite SKUs Info on CTA
* Move connections from Email Management only used with CTA to CTA
* Minor Style Tweaks
* Refactor G Suite Purchase CTA into Functional Component
* Tracks Events when CTA is viewed and "Add G Suite" Button is clicked
* Tests

#### The Component is Visually Unchanged
<img width="500" alt="Current Preview" src="https://user-images.githubusercontent.com/2810519/55842213-5914ee80-5ae7-11e9-818e-0c1082067ba8.png">

#### Preview of both SKUs ( changes not in this PR )
<img width="500" alt="Two SKU Preview" src="https://user-images.githubusercontent.com/2810519/55846780-c599e880-5afb-11e9-876d-f729c57d3697.png">

#### Testing instructions

1. Enable Analytics logging by typing `localStorage.setItem( 'debug', 'calypso:analytics:*' )` into the console of your browser dev tools
1. Make sure your Dev Tools are open so you can confirm Tracks events are correct ( clearing it would probably help )
1. Navigate to `/email/:domain/manage/:siteSlug` for a domain and site without G Suite
1. Confirm that what you see almost exactly matches the WordPress.com version of the same path
1. Confirm that a `calypso_email_gsuite_purchase_cta_view` event was logged with the correct `domain_name`
1. Press "Add G Suite"
1. Confirm you were taken to `/email/:domain/add-gsuite-users/:site`
1. Confirm that a `calypso_email_gsuite_purchase_cta_get_gsuite_click` event was logged with the correct `domain_name` and `plan_type: "basic"`
1. run `npm run test-client client/my-sites/email/gsuite-purchase-cta/` and confirm the tests pass
